In [2]:
import pandas as pd

descriptions = pd.read_csv('dataset/data_descriptions.csv')
train = pd.read_csv('dataset/train.csv')
test = pd.read_csv('dataset/test.csv')

target_col = "Churn"  # change this if your column name is different

# Absolute counts
class_counts = train[target_col].value_counts()
print("Class counts:\n", class_counts)

# Relative proportions
class_ratios = train[target_col].value_counts(normalize=True)
print("\nClass proportions:\n", class_ratios)

# Optional: pretty print
for cls, count in class_counts.items():
    ratio = class_ratios[cls]
    print(f"Class {cls}: {count} samples ({ratio:.2%})")

Class counts:
 Churn
0    199605
1     44182
Name: count, dtype: int64

Class proportions:
 Churn
0    0.818768
1    0.181232
Name: proportion, dtype: float64
Class 0: 199605 samples (81.88%)
Class 1: 44182 samples (18.12%)


our target variable is imbalanced with ~82% non-churners and ~18% churners.

A dataset is considered imbalanced when:

One class > 70%

Minority class < 30%